src/client/utils/dom.ts

```diff
@@ -61,7 +61,9 @@ function isLocationByProto (instance: any): boolean {
         return instance.replace && (isSafari || !!instance.assign);
     }
 
-    return instanceCtor && nativeMethods.objectToString.call(instanceCtor) === '[object LocationPrototype]';
+    return instanceCtor &&
```

Comment:
Store `nativeMethods.objectToString.call(instanceCtor)` value to new variable.

Reply:
```js
if (!instanceCtor)
    return false;

const instanceCtorToString = nativeMethods.objectToString.call(instanceCtor);

return instanceCtorToString === '[object LocationPrototype]' ||
    instanceCtorToString === '[object Location]'; // NOTE: "iPhone" Chrome device emulation case (GH-2080)
```